config-python/config_program.py

```diff
@@ -51,33 +51,35 @@
 NOTIFY_TIMEOUT = 5000
 
 # Public Onboarding Keys
-if os.path.isfile('/var/data/public_keys'):
-    public_keys = {}
-    with open("/var/data/public_keys") as f:
-        for line in f.readlines():
-            # This is insanely ugly, but it gets the
-            # job done until we switch to the API
-            erlang_to_json = line.replace('.', '').\
-                replace(',', ': ').\
-                replace('pubkey', '"pubkey"').\
-                replace('onboarding_key', '"onboarding_key"').\
-                replace('animal_name', '"animal_name"')
-
-            # Let's future proof this just
-            # in case something changes later
-            try:
-                json_line = json.loads(erlang_to_json)
-                for key in json_line.keys():
-                    public_keys[key] = json_line[key]
-            except json.JSONDecodeError:
-                pass
-
-    pubKey = public_keys.get('pubkey', False)
-    onboardingKey = public_keys.get('onboarding_key', False)
-    animalName = public_keys.get('animal_name', False)
-else:
-    print('File public key file not found. Going to sleep')
-    sleep(60)
+while True:
+    if os.path.isfile('/var/data/public_keys'):
+        public_keys = {}
+        with open("/var/data/public_keys") as f:
+            for line in f.readlines():
+                # This is insanely ugly, but it gets the
+                # job done until we switch to the API
+                erlang_to_json = line.replace('.', '').\
+                    replace(',', ': ').\
+                    replace('pubkey', '"pubkey"').\
+                    replace('onboarding_key', '"onboarding_key"').\
+                    replace('animal_name', '"animal_name"')
+
+                # Let's future proof this just
+                # in case something changes later
+                try:
+                    json_line = json.loads(erlang_to_json)
+                    for key in json_line.keys():
+                        public_keys[key] = json_line[key]
+                except json.JSONDecodeError:
+                    pass
```

Comment:
At this point, the file definitely exists, but is just malformed. Maybe this will fix itself in some cases. But in others, there's a permanent error that will cause this to be stuck in an infinite loop. At the minimum, we should log this and potentially even terminate at this point. I will address this in #61

Reply:
The reason why I added this logic was as a safety net in case a new line is added that we can't parse. It would then just silently ignore it since we don't really care about it.